didcomm/types/attachment.py

```diff
@@ -0,0 +1,25 @@
+from typing import NamedTuple, Optional, List
+
+from didcomm.types.types import JWS, JWE, JSON
+
+
+class AttachmentData(NamedTuple):
+    """data property of an attachment."""
+    jws: Optional[JWS] = None
+    jwe: Optional[JWE] = None
+    json: Optional[JSON] = None
+    base64: Optional[str] = None
+    links: Optional[List[str]] = None
+    hash: Optional[str] = None
+
+
+class Attachment(NamedTuple):
+    """attachments list element of a plaintext."""
+    id: str
+    data: AttachmentData
```

Comment:
I think it makes sense to exclude the possiblity to  specify mutually exclusive options for the data, see https://github.com/sicpa-dlab/didcomm-ts/pull/1/files#diff-d79dae8f64b2335dd2f3a749f7691ba91234a557e4809f938b4431c183912f1aR188